katello/hooks/boot/01-helpers.rb

```diff
@@ -44,6 +44,38 @@ def execute_command(command, do_say, do_log)
         $?.success?
       end
     end
+
+    def start_service(name)
+      return execute("foreman-maintain service start --only #{name}") if foreman_maintain?
+
+      if service_available?(name)
+        log_and_say :info, 'Expecting #{name} is managed by Foreman'
+        execute("systemctl start #{name}")
+      else
+        log_and_say :info, 'Expecting #{name} is not managed by Foreman, not trying to start'
+        true
+      end
+    end
+
+    def stop_service(name)
+      return execute("foreman-maintain service stop --only #{name}") if foreman_maintain?
+
+      if service_available?(name)
+        log_and_say :info, 'Expecting #{name} is managed by Foreman'
+        execute("systemctl stop #{name}")
+      else
+        log_and_say :info, 'Expecting #{name} is not managed by Foreman, not trying to stop'
+        true
+      end
+    end
+
+    def service_available?(name)
+      execute("which #{name}")
+    end
+
+    def foreman_maintain?
```

Comment:
This method will be called multiple times, I am not sure where and how I can put it so it gets called only once for installer run. Any suggestions are welcome.